extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/internal/LifxNetworkThrottler.java

```diff
@@ -25,8 +27,8 @@
 
     public final static long PACKET_INTERVAL = 50;
 
-    private static ConcurrentHashMap<String, ReentrantLock> locks = new ConcurrentHashMap<String, ReentrantLock>();
-    private static ConcurrentHashMap<String, Long> timestamps = new ConcurrentHashMap<String, Long>();
+    private static Map<String, ReentrantLock> locks = Collections.synchronizedMap(new HashMap<String, ReentrantLock>());
```

Comment:
This data structure still does not solve the issue that when as the `Map` grows it starts returning the locks in a different order when `values()` is called. All threads should really lock in the same order to get rid of the deadlock.

You can clearly see it happen when you make a simple test printing `String`s as map values. It kicks even in earlier than I thought. Especially when there are a lot of bulbs, the order starts changing a lot each time a new lock is added:

```
    Map<String, String> keyValueMap = Collections.synchronizedMap(new HashMap<String, String>());

    @Test
    public void valuesTest()
    {
        String values = "";

        for (int i = 0; i < 10; i++)
        {
            keyValueMap.put("key" + i, "value" + i);
            values = StringUtils.join(keyValueMap.values(), ",");

            System.out.println(values);

        }
    }


The output of system is then:

value0
value1,value0
value1,value2,value0
value1,value2,value0,value3
value1,value2,value0,value3,value4
value1,value2,value0,value5,value3,value4
value1,value2,value0,value5,value6,value3,value4
value1,value2,value0,value5,value6,value3,value4,value7
value1,value2,value0,value5,value6,value3,value4,value7,value8
value1,value2,value0,value5,value6,value3,value4,value9,value7,value8
```